### PR TITLE
[Remote Store] Add check for byte lag in time lag calculation

### PR DESCRIPTION
**Pull request by ashking94**
### Description
In this PR, we are adding a defensive check which is used for determining the time lag. The time lag can show erroneous value if there is a refresh without any additional data ingested as segments.

### Related Issues
No issue created.

### Check List
- ~[ ] New functionality includes testing.~
  - [x] All tests pass
- ~[ ] New functionality has been documented.~
  - ~[ ] New functionality has javadoc added~
- [x] Failing checks are inspected and point to the corresponding known issue(s) (See: [Troubleshooting Failing Builds](../blob/main/CONTRIBUTING.md#troubleshooting-failing-builds))
- [x] Commits are signed per the DCO using --signoff
- ~[ ] Commit changes are listed out in CHANGELOG.md file (See: [Changelog](../blob/main/CONTRIBUTING.md#changelog))~
- ~[ ] Public documentation issue/PR [created](https://github.com/opensearch-project/documentation-website/issues/new/choose)~

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: server/src/test/java/org/opensearch/index/remote/RemoteSegmentTransferTrackerTests.java [hunk 1]
```diff
@@ -149,6 +150,7 @@ public void testComputeTimeLagOnUpdate() throws InterruptedException {
         Thread.sleep(1);
         transferTracker.updateLocalRefreshTimeMs(currentTimeMsUsingSystemNanos());
 
+        transferTracker.updateLatestLocalFileNameLengthMap(List.of("test"), k -> 1L);
```
Comment (sachinpkale) [inline server/src/test/java/org/opensearch/index/remote/RemoteSegmentTransferTrackerTests.java:153]: Is this related to change in this PR?

**Review by gbbafna — APPROVED** [review]
